File: lua/todo.lua
local tasks = {}

local function add_task(task)
    table.insert(tasks, task)
    print("Task added.")
end

local function list_tasks()
    for i, task in ipairs(tasks) do
        print(i .. ": " .. task)
    end
end

local function remove_task(index)
    if index >= 1 and index <= #tasks then
        table.remove(tasks, index)
        print("Task removed.")
    else
        print("Invalid index.")
    end
end

if #arg < 1 then
    print("Usage: lua todo.lua add \"task\" | list | remove <index>")
    os.exit(1)
end

local command = arg[1]
if command == "add" and #arg == 2 then
    add_task(arg[2])
elseif command == "list" then
    list_tasks()
elseif command == "remove" and #arg == 2 then
    remove_task(tonumber(arg[2]))
else
    print("Invalid command.")
end

File: ruby/todo.rb
tasks = []

def add_task(task)
  tasks << task
  puts "Task added."
end

def list_tasks
  tasks.each_with_index { |task, i| puts "#{i + 1}: #{task}" }
end

def remove_task(index)
  if index.between?(1, tasks.size)
    tasks.delete_at(index - 1)
    puts "Task removed."
  else
    puts "Invalid index."
  end
end

if ARGV.empty?
  puts "Usage: ruby todo.rb add \"task\" | list | remove <index>"
  exit 1
end

command = ARGV[0]
case command
when "add"
  add_task(ARGV[1]) if ARGV.size == 2
when "list"
  list_tasks
when "remove"
  remove_task(ARGV[1].to_i) if ARGV.size == 2
else
  puts "Invalid command."
end
